### ISSUE TITLE
drawing operations performed twice during startup

### ISSUE THREAD

**Issue by dkochmanski** [McCLIM/McCLIM#937]
This is a result of lack of coordination between the repaint protocol and the stream output recording protocol. Typical scenario for output recording stream:

applicaiton starts
window is opened
stream drawing operations happen | window repaint event queued (port thread)
stream-replay
window repaint event triggers handle-repaint and in consequence stream-replay

Default method handle-repaint on seos (standard-extended-output-stream) does nothing unlike the sors (standard-output-recording-stream), that's why this is more pressing for recording streams.

**Comment by dkochmanski** [comment]
Issue is impossible to solve completely because of asynchronous nature of events. window-repaint-event may come at any time. solution I've taken: displaying the function first records the output, then dispatches a repaint. We have now three scenarios (we are assuming standard-repaint-mixin in this analysis):

1. window-repaint-event is in the queue, then the new event is compressed with it. during next event read window is repainted
2. there is no window-repaint-event in the queue, then our dispatched event is *the* event
2a. a new window-repaint-event comes before event read: it is compressed
2b. event is read, then window-repaint-event comes, window is repainted twice

in case of 2b it is impossible to say if the region was damaged after the last repaint or before it, hence we need to repaint to be sure.

in other words the change (in the upcoming pr) makes it less likely that sheet is repainted twice, however it may still happen.

Detecting if we are during the repaint and refuse to add window-repaint-event if it repaints a subregion of the current repaint doesn't make sense either, because:
1. we replay region a
2. we replay region b
3. region a is damaged
4. window-repaint-event for region a is sent
5. we replay region c

if we ignore the event, a is damaged.
